scripts/pipelines/gcloud/templates/package/package-pipeline.cfg

```
@@ -0,0 +1,144 @@
+    then
+      for i in $dockerfilePaths
+      do
+        echo $i | grep "$language:" > /dev/null && dockerFile=$(echo $i | cut -d':' -f2) && break
+      done
+    fi
+
+    # We get the name of the registry from the full image name
+    firstPartImage=$(echo $imageName | cut -d'/' -f1)
+    echo $firstPartImage | grep "\." > /dev/null && registry=$firstPartImage || registry="docker.io"
+
+    if ! gcloud secrets describe registry-credentials &> /dev/null
+    then
+        gcloud secrets create registry-credentials --replication-policy="automatic"
+    fi
+
+    if test -z "$aws_access_key"
+    then
+       echo -e "DOCKER_USER=$dockerUser\nDOCKER_PASSWORD=$dockerPassword" | gcloud secrets versions add registry-credentials --data-file=-
+    else
+       echo "AWS_ACCESS_KEY=$awsAccessKey\nAWS_SECRET_ACCESS_KEY=$awsSecretAccessKey\nAWS_REGION=$awsRegion" | gcloud secrets versions add registry-credentials --data-file=-
+    fi
+
+    # Setting the list of variable that will be used for this trigger
+    subsitutionVariable="_DOCKER_IMAGE=${dockerImage},_DOCKERFILE=${dockerFile},_REGISTRY=${registry},_IMAGE_NAME=${imageName},_BUILD_PIPELINE_NAME=${buildPipelineName},_BRANCH_NAME=\$BRANCH_NAME"
+}
+
+# Function that prepares Artifact Repository for storing the docker image
+function createArtifactRegistry {
+    # When using GCP Artifact Registry, make sure that the Docker repository exists before pushing to it, if not, create it
+    repositoryName=$(echo "$imageName" | cut -d '/' -f 3)
+    region=$(echo "$imageName" | cut -d '/' -f 1 | sed 's/-docker.pkg.dev//' )
+    if ! gcloud artifacts repositories describe "$repositoryName" --location="$region" &> /dev/null
+    then
+        # We create the Docker repository in Artifact Registry
+        gcloud artifacts repositories create "$repositoryName" --repository-format=docker --location="$region"
+    fi   
+}
+
+if [[ "$imageName" == *"docker.pkg.dev"* ]] 
+then
+    type createArtifactRegistry &> /dev/null && createArtifactRegistry
+fi
+
+# Function that prepares a bucket for storing the .apk artifact
+function createBucketforApk {
+    oldPath="$PWD"
+    cd "$localDirectory"
+    gitOriginUrl=$(git config --get remote.origin.url)
+    gCloudProject=$(echo "$gitOriginUrl" | cut -d'/' -f5)
+    export gCloudProject
+    gCloudRepo=$(echo "$gitOriginUrl" | cut -d'/' -f7)
+    # We check if the bucket we need exists, we create it if not
+    if (gcloud storage ls --project="$gCloudProject" | grep "$gCloudProject-apk" >> /dev/null)
+    then
+        echo -e "${green}Bucket $gCloudProject-apk already exists.${white}"
+    else
+        echo -e "${green}The bucket $gCloudProject-apk does not exist, creating it...${white}"
+        if ! gcloud storage buckets create gs://"$gCloudProject"-apk --project="$gCloudProject"
+        then
+            echo -e "${red}Error: While creating buckeet for $gCloudProject-apk." >&2
+            echo -ne "${white}" >&2
+            exit 2
```

```suggestion
            exit 1
```

Not bad argument, right?

---

you are right